text_collector_examples/textfile_wrapper.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#
+# Copyright 2018 The Prometheus Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Description: Wrapper script to atomically write prometheus textfile script output.
+
+set -u
```

Comment:
I'd also add -e

As-is, if the script fails the output is still used.

Reply:
I'm a big un-fan of bash `set -e`.  I'd much rather catch any problems explicitly, since bash exits somewhat silently on `set -e`.